proposals/0000-improved-ghc-development.rst

```diff
@@ -0,0 +1,216 @@
+  both of which seem to be quite bare and simple in what they provide. I am not
+  in favor of ditching it. Whatever we gain we lose in the ability to generate custom views on issues. Which is a powerful filtering mechanism for prioritization.
+
+  Also we have and continue to accept patches just uploaded
+  to Trac as a diff. We tend to ask people to upload it to phab for better reviews
+  and so it's attributed to them when we commit. Some don't (and we then do it ourselves),
+  most due. If they don't need another login then I suspect almost all would.
+
+  There's a (seemingly) actively maintained project that does all the above, could we leverage it?
+  https://github.com/trac-hacks/trac-github
+
+- Allow doc changes on Trac
+
+  I am not sure about the feasibility of this one, but there is a Trac plugin that adds a new section to trac under /doc that is able to render .rst files and more from a source repository. Essentially making a wiki from them.
+
+  https://github.com/trac-hacks/tracdocs
+
+  We could have harbormaster build the docs and provide them to trac. I am not sure how reviews would go. Styling would be another issue. 
+
+  Lastly this plugin seems to only support SVN. but that's not a huge barrier.
+
+- Helping Newcomers
+
+   * Create a better landing page for newcomers (see proposal below)
+   * Show newcomers tickets more prominently
+   * Clean up build instructions. The pages have become way too complex over the years. 
+   * Create native setup script particular to an environment.
+
+  As I am a windows maintainer, I will offer a proposal: Use scripts similar to my own setup scripts. my setup script for a 100% unattended build env setup for Windows are here: https://github.com/Mistuke/GhcDevelChoco/releases
+
+  These are entirely self contained environments that can be removed by a simple rm -rf /.
+  You can have as many as you want on the same machine without them interfering with eachother or with whatever else you might have done to your GHC already installed.
+   
+  It's not 100% production ready but it works and does so well.
+
+  This script will setup an environment containing (if you want):
+   * A full msys setup
+   * gdb/gcc etc for debugging
+   * check out Haskell sources
+   * do a local install of php and arc
+   * configure arc
+   * setup Hadrian
+   * setup SSH (for those who want to interface with it this way)
+ 
+  The script is set up to provide you with a working environment with little interaction needed. Setup is just one call. 
+
+  For Windows, such a script is a much better solution than Docker, simply because Docker does not work very well for Windows targets.
+
+- Phab reviewers list
+
+  Reviewers are currently assigned based on a set of Phabricator rules. As a maintainer on a less popular platform I know the pain of getting people to review your code.
+  
+  So my suggestion is to also assign reviewers based on the git history of the files you're changing. 
```

Comment:
This has been proposed upstream in Phabricator too but honestly I think it's messy and heuristic to work out. What's the threshold for reviewers, how much of the file they wrote? How many commits they have in 6 months to that file? How many lines they changed? It's extremely hard to determine what's "significant" just from a file log... Many rote or fundamental changes may cause simple mechanical refactorings that touch these files, but that doesn't necessarily imply a culprit for breakage, or that the user in question is appropriate to review other things touching that file.

I think a much better heuristic is to "scale" the Phabricator rules, using the "Owners" application. This means all regular committers can assign their "ownership" over a particular set of files, and when a review comes in and it touches those files, they're added. The hope here would be there'd at least be one owner covering every component who has to sign off on something. This would require "codifying" the sort of code owners concept we have now, but enforcing that may be better in the long run.